### PR TITLE
Custom Range: Analytics

### LINKED ISSUE #11050: Analytics

### PR DESCRIPTION
**Pull request by 0nko**
Closes #11050. The PR adds tracking to the following events:

```
DASHBOARD_STATS_CUSTOM_RANGE_ADD_BUTTON_TAPPED
DASHBOARD_STATS_CUSTOM_RANGE_CONFIRMED
DASHBOARD_STATS_CUSTOM_RANGE_TAB_SELECTED
DASHBOARD_STATS_CUSTOM_RANGE_EDIT_BUTTON_TAPPED
DASHBOARD_STATS_CUSTOM_RANGE_INTERACTED
```

**To test:**
1. Start the app and add a custom range
2. Change the tabs
3. Edit the range
4. Verify the range button, tab selection, range edit and confirmation are tracked
5. Pay special attention to the interactions events `DASHBOARD_STATS_CUSTOM_RANGE_INTERACTED` when tapping on the graph and the custom range is selected
6. Change the tab to a non-custom range
7. Verify the `USED_ANALYTICS` events is tracked instead

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/203">18.0</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false"><p>This PR contains changes to Tracks-related logic. Please ensure (<strong>author and reviewer</strong>) the following are completed:</p>

<ul>
  <li>The tracks events must be validated in the Tracks system.</li>
  <li>Verify the internal Tracks spreadsheet has also been updated.</li>
  <li>Please consider registering any new events.</li>
  <li>The PR must be assigned the <strong>category: tracks</strong> label.</li>
</ul></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by JorgeMucientes** [comment]
Nice job @0nko. 

While testing this I found a couple of issues. The second is unrelated to these PR changes: 
1. I wasn't able to trigger `dashboard_stats_custom_range_interacted` by interacting with the chart. The only way I got this to track `dashboard_stats_custom_range_interacted` was after killing the app and the clicking on custom range tab. I think adding it inside the `MyStoreStatsUsageTracksEventEmitter.interacted()` is not going to work as we expect this event to work: 
    > When the user taps on the chart on custom range to see metrics for specific periods.

    I think we'd need to rethink where we add this event and how we avoid not sending it to many times.

2. I think there's a problem with `getEntryValuesForCustomType()` that we are tracking `stats_unexpected_format` all the time.  I will look into this because it seems its a bug I introduced when refactoring MyStoreStatsView

EDIT: I fixed the second issue in this PR https://github.com/woocommerce/woocommerce-android/pull/11180

**Comment by 0nko** [comment]
@JorgeMucientes Thank you for the review. The `MyStoreStatsUsageTracksEventEmitter` contains a special logic for the way interactions are measured. I haven't changed this logic itself, so interactions with custom range should behave the same as other ranges. 

I tried adding breakpoints to the event trigger and I was able to get the `dashboard_stats_custom_range_interacted` triggered, just like `used_analytics` when using a regular range. Bear in mind that you need to interact with the graph at least 5 times within 10 seconds. 

If the algorithm is broken for custom range, it's broken for the rest as well 🤷.

**Comment by JorgeMucientes** [comment]
Hey @0nko. 

> The MyStoreStatsUsageTracksEventEmitter contains a special logic for the way interactions are measured. I haven't changed this logic itself, so interactions with custom range should behave the same as other ranges.

Yeah, I saw you didn't change that logic. My question is whether we should track `dashboard_stats_custom_range_interacted` inside `MyStoreStatsUsageTracksEventEmitter.interacted()`. I'm not sure it belongs there. Doing so will not make the event behave as defined in the PT tracking section: `When the user taps on the chart on custom range to see metrics for specific periods.` It won't behave like iOS does either, where this event is tracked whenever we tap on the chart or lift up our finger after sliding through the chart. 

Moreover, adding this condition inside `MyStoreStatsUsageTracksEventEmitter.interacted()`: 
```kotlin
            if (isCustomRange) {
                analyticsTrackerWrapper.track(AnalyticsEvent.DASHBOARD_STATS_CUSTOM_RANGE_INTERACTED)
            } else {
                analyticsTrackerWrapper.track(AnalyticsEvent.USED_ANALYTICS)
            }
```
will have an unexpected side effect of reducing the number of times `analyticsTrackerWrapper.track(AnalyticsEvent.USED_ANALYTICS)` is tracked if the custom range is selected. 

Imho I'd handle the tracking of `dashboard_stats_custom_range_interacted` directly from `MyStoreStatsView`. 

I don't think this is a blocking issue though, so I'll pre-approve the PR. But let me know what you think of this :)

**Review by JorgeMucientes — APPROVED** [review]

**Comment by 0nko** [comment]
> Imho I'd handle the tracking of dashboard_stats_custom_range_interacted directly from MyStoreStatsView.

Oh, I see what you mean now... I'll try something different, then..

**Comment by 0nko** [comment]
@JorgeMucientes I've updated the tracking logic. The tracking should now match the iOS behavior more closely and the original `USED_ANALYTICS` is not affected anymore.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/mystore/MyStoreViewModel.kt [hunk 1]
```diff
@@ -231,6 +231,12 @@ class MyStoreViewModel @Inject constructor(
         usageTracksEventEmitter.interacted()
         _selectedRangeType.update { selectionType }
         appPrefsWrapper.setActiveStatsTab(selectionType.name)
+
+        if (isCustomRange) {
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/mystore/MyStoreViewModel.kt]:
```suggestion
        if (selectionType == SelectionType.CUSTOM) {
```

Minor thing. After the refactor, the code doesn't compile unless I add this.

**Review by JorgeMucientes — APPROVED** [review]

**Comment by JorgeMucientes** [comment]
Thank you for handling the changes, @0nko. It works as expected now and doesn't have side effects over the existing tracking 👍🏼 
I just left a minor suggestion that is needed for the code to compile. Other than that is ready to 🚢 !

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/mystore/MyStoreViewModel.kt (hunk 1 above)
Comment (0nko) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/mystore/MyStoreViewModel.kt]: Oops, that was an artefact of a reversed commit. Thanks!

**Pull request merged**